index/scorch/merge.go

```diff
@@ -167,4 +168,72 @@ type segmentMerge struct {
 	oldNewDocNums map[uint64][]uint64
 	new           segment.Segment
 	notify        notificationChan
+	notifyResult  chan *IndexSnapshot
```

Comment:
is there any way to unify the behavior?  it seems like disk merge uses notify, and that didn't work for you, you added your own notifyResult, but we also inherited a bunch of conditional logic to work with both of them now

Reply:
> is there any way to unify the behavior?

yah, will unify.  The reason I didn't unify them before was that the new merge situation needed another AddRef() on the IndexSnapshot when passing it around..., so that felt a little different.  Wasn't brave enough to change the various places as part of this.

Reply:
Can you elaborate on why the reference counting ends up being different?

Reply:
> Can you elaborate on why the reference counting ends up being different?

Not different per se, but actually must be used in the "notifyResult chan *IndexSnapshot".

That is, when the introduceMerge() wants to sent an IndexSnapshot through the notifyResult chan, it first has to AddRef() it.  And, the receiver reading the chan has to DecRef() it.

I was leery of introducing that ref-counting tax to the existing "notify" codepaths.

Actually, on noodling about this on the drive into the cafe, I now a bit unsure on what's the right path.